EddieHubCommunity/LinkFree
# fix: improve error handling  with HTTP utility functions

## Fixes Issue
Closes #8112 






## What?
added a new util file `apiRequests.js`, containing functions for sending API requests.
Implemented `sendRequest` function that handles various types of API requests, handles rejected promise and also converts the `response` to `JSON`

## Why? 
abstract `api` request functionality into one function that we can use all over our project.
These changes improved error handling

## How?
replace `fetch` with `sendRequest` in manage profile sections

## Testing? 
**Step to test**
* Pull down this branch
* visit `localhost:3000/account/manage/profile` (link, milestone, event, testimonials) 
* send an invalid api request by `sendRequest` function
* see the request actually rejected and displayed the error to UI

## Anything else?
Let's consider implementing it other areas of the project. I will happy to doing that work💪






## Check List (Check all the applicable boxes) 

- [ ] My change requires changes to the documentation.

## Screenshots



## Note to reviewers




<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8283"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
